browser/src/control/Control.QuickFindPanel.ts

```diff
@@ -36,23 +36,45 @@ class QuickFindPanel extends SidebarBase {
 
 		super.onJSUpdate(e);
 
-		// handle placeholder text visibility
+		// handle placeholder text and quickfind controls visibility
 		app.layoutingService.appendLayoutingTask(() => {
-			const placeholder = document.getElementById('quickfind-placeholder');
-
 			if (data.control.id === 'numberofsearchfinds') {
-				const resultsText = data.control.text.trim().length > 0;
-				if (placeholder) placeholder.classList.toggle('hidden', resultsText);
+				this.handleNumberOfSearchFinds(data.control);
 			} else if (
 				data.control.id === 'searchfinds' &&
 				data.control.type === 'treelistbox'
 			) {
-				const isEmpty = data.control.entries.length === 0;
-				if (placeholder) placeholder.classList.toggle('hidden', !isEmpty);
+				this.handleSearchFindsTreelistbox(data.control);
 			}
 		});
 	}
 
+	handleNumberOfSearchFinds(control: any): void {
+		const placeholder = document.getElementById('quickfind-placeholder');
+		const resultsLabel = document.getElementById(
+			'numberofsearchfinds',
+		) as HTMLLabelElement;
+
+		const hasText = (control.text ?? '').trim().length > 0;
+		if (placeholder) placeholder.classList.toggle('hidden', hasText);
+		if (hasText && control.text === '0 results')
```

Comment:
Unfortunetely: we will receive translated string, you cannot just compare with English version.
Could you just use the knowledge from a tree widget where there will be 0 entries?

Reply:
Removed the label modification as it will require to be changed in core